examples/Example/src/prot_calculator/index.tsx

```diff
@@ -0,0 +1,354 @@
+import React from 'react';
+import { useRef } from 'react';
+import { useState } from 'react';
+import { StyleSheet, View, Text, Dimensions } from 'react-native';
+import {
+  GestureMonitor,
+  Gesture,
+  ScrollView,
+} from 'react-native-gesture-handler';
+import Animated, {
+  useSharedValue,
+  useAnimatedStyle,
+  withTiming,
+  runOnJS,
+} from 'react-native-reanimated';
+import { useAnimatedGesture } from '../useAnimatedGesture';
+
+const window = Dimensions.get('window');
+
+export default function CalculatorUI() {
+  const outputOffset = useSharedValue(0);
+  const [history, setHistory] = useState([]);
+  const [expression, setExpression] = useState('');
+
+  function measure(e) {
+    outputOffset.value = -e.nativeEvent.layout.height;
+  }
+
+  return (
+    <View style={styles.home}>
+      <Output offset={outputOffset} expression={expression} history={history} />
+      <Input
+        offset={outputOffset}
+        measure={measure}
+        expression={expression}
+        setExpression={setExpression}
+        setHistory={setHistory}
+      />
+    </View>
+  );
+}
+
+function Output(props) {
+  const layout = useRef({});
+  const scrollView = useRef();
+  const drag = useSharedValue(0);
+  const [opened, setOpened] = useState(false);
+
+  function measure(e) {
+    layout.current = e.nativeEvent.layout;
+  }
+
+  function open() {
+    drag.value = withTiming(-props.offset.value, { duration: 300 });
```

Comment:
Duration should be extracted to constant, you're using it in few places in this file.